### DISCUSSION TITLE
Why the channel is closed when using QueueInspect?

### DISCUSSION THREAD

**Discussion by gabo1208** [rabbitmq/amqp091-go#165] [General]
I use the channel.QueueInspect method to check if a Queue exists but I found annoying that it closes the channel every time I do, is there any reason why it is this way? Or it could be improved by collaborating =)?

**Comment by lukebakken** [comment]
https://github.com/rabbitmq/amqp091-go/blob/main/channel.go#L866-L896

`QueueInspect` assumes that no special declaration parameters have been specified for a queue. My guess is that RabbitMQ logs a "Precondition Failed" error when you use `QueueInspect`.

So, if you call `QueueInspect` on an exclusive queue, it will fail.

I was unaware of `QueueInspect` until now, and it seems like a good candidate for deprecation and removal in 2.0. Thanks.

**Comment by Zerpet** [comment]
I recommend that you use `Channel.QueueDeclarePassive` instead. This function `QueueInspect` is using a passive queue declaration, but without giving you control over the arguments, which is likely to return an error if any of ther queue parameters and/or argument don't match.

How extensible do you use this function? How bad will be the impact if we deprecate it?

For context, this function is not the AMQP spec, so it was probably added as a convenience function, that would work most of the time, when there was only one queue type.

**Reply by gabo1208** [reply]
In a few places but nothing too big, so go ahead, I'll be working on using `Channel.QueueDeclarePassive` instead (that I do not knew). So thanks =)
